src/helpers/model/measureValueFilters.ts

```diff
@@ -0,0 +1,116 @@
+// (C) 2019 GoodData Corporation
+import { AFM } from "@gooddata/typings";
+import {
+    IValue,
+    IMeasureValueComparisonFilter,
+    IMeasureValueRangeFilter,
+    isRangeTypeOperator,
+    isComparisonTypeOperator,
+} from "../../components/filters/MeasureValueFilter/types";
+import { getQualifier } from "./utils";
+
+export class MeasureValueComparisonFilterBuilder implements IMeasureValueComparisonFilter {
+    public measureValueFilter: IMeasureValueComparisonFilter["measureValueFilter"];
+
+    constructor(identifier: string, operator: AFM.ComparisonConditionOperator = null, value: IValue = null) {
+        this.measureValueFilter = {
+            measure: getQualifier(identifier),
+            condition: {
+                comparison: {
+                    operator,
+                    value: value.value || 0,
+                },
+            },
+        };
+    }
+
+    public setOperator(operator: AFM.ComparisonConditionOperator) {
```

Comment:
I think the getters and setters are not needed here. I'd prefer making the corresponding parameters of the constructor mandatory as the resulting object does not make sense without operator and value. The idea behind the builder methods was that any time a user creates an instance using the constructor, the resulting object is valid and makes sense. The methods are then used only to set optional parameters (e.g. alias, title).

That being said, if you remove these, the builder class will no longer make sense, as it is only a constructor. Therefore, I'd delete it, and reimplement `getComparisonFilter` like

```ts
function getComparisonFilter(
    identifier: string,
    operator: AFM.ComparisonConditionOperator,
    value: IValue,
): IMeasureValueComparisonFilter {
    return {
        measure: getQualifier(identifier),
        condition: {
            comparison: {
                operator,
                value,
            },
        },
    };
}
```

The same is applicable to the `MeasureValueRangeFilterBuilder` class and its respective `getRangeFilter` function.

Reply:
I have also added an operator and value getter and few type guards after discussion.